### PR TITLE
Add the macro command

### PR DESCRIPTION
**Pull request by Juliacare**
Adding a new macro command to quickly and reliably display information of frequently asked queries.

Finally, retire p!time as well.

### CODE REVIEW

File: commands/general/macro.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+module.exports.load = (client) => {
+  client.commands['macro'] = {
```
Comment (sit) [inline commands/general/macro.js]: I don't love the name `macro`. These feel sort of like sub-commands of `help` to me. Any other names come to mind?

File: commands/general/macro.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            msg.addField('Leaderboards', `
+                  \`Pinano Bot\` - The pinano bot records how long you practice each day!
+                  \`Leaderboards\` - Compete with others for a spot on the weekly or overall leaderboards!
+                  \`Earn ranks\` - Earn \`Active Pracker\` or \`40 Hour Pracker\` roles by practicing on voice chat.
+                  \`Commands\` - Use p!lb overall, p!lb weekly or p!lb stats to see how you're doing!
+                  `, false)
+            msg.setColor(client.settings.embed_color)
+            msg.setTimestamp()
+
+  //    break
```
Comment (sit) [inline commands/general/macro.js]: I'd leave this `break` here.

File: commands/general/macro.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+  //      case 'recital':
```
Comment (sit) [inline commands/general/macro.js]: I generally don't like committing commented code. This seems pretty easy to add when we're ready to add it, so can you just delete it?

**Comment by sit** [comment]
Might need to be rebased, but I'm down to merge this. We can change things like the name later if we want.

Long term, it will probably be nice to have the text read from files that are not directly embedded in source (e.g. as YAML or Markdown or plain text). That way people who aren't coders will be able to modify the text easily.

File: commands/general/macro.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            msg.setTimestamp()
+
+      break
+
+        case 'chat':
+            let msg = new client.discord.RichEmbed()
+            msg.setTitle('Discord chat channels information')
+            msg.setDescription( 'Heres a list of our chat channels' )
+            msg.addField('Chat channels', `
+                  \`#Announcement\` - Stay up to date with the latest Pinano news with our announcements channel!
```
Comment (pianiststickman) [inline commands/general/macro.js]: Be prepared for this to be regularly out of date -_-

File: commands/general/macro.js (hunk 1 above)
Comment (pianiststickman) [reply commands/general/macro.js]: Maybe `info`?

File: commands/general/macro.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      switch (args[0]) {
+        case 'help':
+            let msg = new client.discord.RichEmbed()
+            msg.setTitle('Macro help')
+            msg.setDescription( 'Heres a list of all available macros' )
+            msg.addField('Commands', `
+                  \`${client.settings.prefix}macro help\` - This message.
+                  \`${client.settings.prefix}macro chat\` - Chat channels information.
+                  \`${client.settings.prefix}macro vc\` - Voice chat and practice information.
+                  \`${client.settings.prefix}macro recital\` - Soon (tm)
```
Comment (pianiststickman) [inline commands/general/macro.js]: ™

File: commands/general/macro.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    run (message) {
+      let args = message.content.split(' ').splice(1)
+      switch (args[0]) {
+        case 'help':
+            let msg = new client.discord.RichEmbed()
+            msg.setTitle('Macro help')
+            msg.setDescription( 'Heres a list of all available macros' )
+            msg.addField('Commands', `
+                  \`${client.settings.prefix}macro help\` - This message.
+                  \`${client.settings.prefix}macro chat\` - Chat channels information.
```
Comment (pianiststickman) [inline commands/general/macro.js]: We could have multiple ways of referring to this.

`p!info text`, `p!info text_channels`, `p!info tc` could all be used to trigger the same logic - we'd just need to add it to the switch.

Then the corresponding voice commands would be `p!info voice`, `p!info voice_channels`, and `p!info vc`.

File: commands/general/macro.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+      break
+
+        case 'chat':
+            let msg = new client.discord.RichEmbed()
+            msg.setTitle('Discord chat channels information')
+            msg.setDescription( 'Heres a list of our chat channels' )
+            msg.addField('Chat channels', `
+                  \`#Announcement\` - Stay up to date with the latest Pinano news with our announcements channel!
+                  \`#General\` - Chat and get to know others in our general chat channel!
```
Comment (pianiststickman) [inline commands/general/macro.js]: Nit: can these be lower-cased as on the server?

File: commands/general/macro.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                  \`#Slow Practice Room\` - Taking practicing slow? Just getting started? You can do that here!
+                  \`#Practice room\` - Do you want to share your practicing with others? Let us listen in!
+                  \`#Other instrument room\` - Do you feel like practicing a instrument other than piano? You can do that here!
+                  \`#Recital\` - Want to listen in to the Bi weekly recitals? Join us when the recital channel is open!
+                  `, false)
+            msg.addField('Leaderboards', `
+                  \`Pinano Bot\` - The pinano bot records how long you practice each day!
+                  \`Leaderboards\` - Compete with others for a spot on the weekly or overall leaderboards!
+                  \`Earn ranks\` - Earn \`Active Pracker\` or \`40 Hour Pracker\` roles by practicing on voice chat.
+                  \`Commands\` - Use p!lb overall, p!lb weekly or p!lb stats to see how you're doing!
```
Comment (pianiststickman) [inline commands/general/macro.js]: I think you want `p!stats` here.

**Comment by Juliacare** [comment]
Directly uploaded to not have to deal with conflicting files during a merge

**Pull request closed without merging**